### PR TITLE
Implemented the == operator and hashCode.

### PR DESCRIPTION
**Pull request by nullrocket**

### CODE REVIEW

File: lib/src/json_pointer.dart [hunk 1]
```diff
@@ -103,6 +103,14 @@ abstract class JsonPointer {
   @override
   String toString();
 
+  /// Compares the pointers paths for equality
```
Comment (f3ath) [inline lib/src/json_pointer.dart:106]: Why do we need to redefine these in the interface?

**Comment by f3ath** [comment]
@nullrocket thanks for the PR. I have fixed the format check, could you please rebase on `main`?

File: lib/src/json_pointer.dart (hunk 1 above)
Comment (nullrocket) [reply lib/src/json_pointer.dart]: It doesn't need them,  I started out implementing both methods in the abstract class and didn't remember to remove them when I moved them to JsonPointerSegment.   

There is a toString override already there in the interface that could be removed, it was already there so I left it alone.

Comment (nullrocket) [reply lib/src/json_pointer.dart]: I pushed a new commit cleaning them up.

File: test/equality_test.dart [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  test('Pointer equality', () {
+    var thisPointer = JsonPointer('/root');
+    var thatPointer = JsonPointer('/root');
+    var differentPointer = JsonPointer('/different/one');
+    var explicitEmptyPointer = EmptyJsonPointer();
+    var emptyPointer = JsonPointer();
+
+    expect(thisPointer == thatPointer, true);
+    expect(emptyPointer == explicitEmptyPointer, true);
+    expect(thisPointer != differentPointer, true);
```
Comment (f3ath) [inline test/equality_test.dart:15]: Sorry, I had to bump the dependencies. Would you please rebase once again. Also, let's add some assertions for the `.hashCode`:
```dart
    expect(thisPointer.hashCode == thatPointer.hashCode, true);
    expect(emptyPointer.hashCode == explicitEmptyPointer.hashCode, true);
    expect(thisPointer.hashCode != differentPointer.hashCode, true);
```

**Comment by f3ath** [comment]
@nullrocket The CI is failing due to `.hashCode` line being untested. I've updated dependencies, now it will tell which LoC it's unhappy about. Thanks for your patience :)

**Comment by f3ath** [comment]
Almost there. We just need the coverage https://github.com/f3ath/rfc-6901-dart/pull/2#discussion_r816459949

**Pull request merged**